src/lib.rs

```diff
@@ -177,30 +180,49 @@ impl Level {
 
     #[cfg(feature = "bzip2")]
     fn into_bzip2(self) -> bzip2::Compression {
+        let fastest = bzip2::Compression::fast();
+        let best = bzip2::Compression::best();
+
         match self {
-            Self::Fastest => bzip2::Compression::fast(),
-            Self::Best => bzip2::Compression::best(),
-            Self::Precise(quality) => bzip2::Compression::new(quality.max(1).min(9)),
+            Self::Fastest => fastest,
+            Self::Best => best,
+            Self::Precise(quality) => bzip2::Compression::new(
+                quality
+                    .try_into()
+                    .unwrap_or(0)
+                    .max(fastest.level())
+                    .min(best.level()),
```

Comment:
avoids the counter-intuitive max-min clamping

```suggestion
                    .clamp(fastest.level(), best.level())
```